### ISSUE TITLE
Pagination doesn't work well, isn't integrated to Django 1.7's native pagination, and is poorly documented.

### ISSUE THREAD

**Issue by ursomniac** [etianen/django-watson#117]
My comment is the title.   I really appreciate how well django-watson works and how simple it is to integrate into existing Django code (compared to the other offerings out there).  However, pagination is well-developed in Django 1.7 and it appears that the only way to get it to work under django-watson is to use the default SearchView with GET, but there's no documentation as to how one is able to create navigation around the pagination.  Also, it doesn't appear that overriding view methods (e.g., get_context_data) works properly (though I did see that SearchView is a subclass of ListView, so perhaps I'm missing something).

It would be great if this could be address for the next release.   I was hoping to begin to incorporate django-watson into other projects, and pagination support would make that far more possible.

**Comment by carltongibson** [comment]
@ursomniac Have you read the docs on [Searching Models](https://github.com/etianen/django-watson/wiki/Searching-models)? If so, what's to stop you just passing the result of `watson.search` to a `Paginator` object, pretty much as you would normally?

**Comment by carltongibson** [comment]
Further to that, what's not working, or not documented? [Built-In Views](https://github.com/etianen/django-watson/wiki/Built-in-views) seems to cover everything you need.

**Comment by carltongibson** [comment]
`SearchView` uses Django's own pagination logic, from `generics.ListView`. <https://github.com/etianen/django-watson/blob/master/src/watson/views.py#L85> — I think you should review [Django's Pagination Docs](https://docs.djangoproject.com/en/1.8/topics/pagination/) — I don't think, as it stands, this is an issue with Django Watson

**Comment by etianen** [comment]
Thanks for the back-up! I'm also confused about what's not working here!

**Comment by ursomniac** [comment]
I'll try to put together a complete example.  I did try using the native 1.7 pagination but couldn't get the Pagination variables (is_paginated, etc.) to be set properly.   But knowing that the watson.search object should be passable to a native Pagination object is a help --- I must have overlooked SOMETHING (but I didn't see anything obvious when I was debugging).

**Comment by ursomniac** [comment]
I do note that "Searching Models" doesn't mention pagination at all.   "Built-In Views" documents the paginate_by attribute, but doesn't say anything about how that affects anything else.

**Comment by etianen** [comment]
The SearchView is a django.views.generic.ListView. It looks like the django documentation around using paginate_by in ListView is pretty well hidden, but here's the relevant links:

https://docs.djangoproject.com/en/1.8/ref/class-based-views/mixins-multiple-object/#django.views.generic.list.MultipleObjectMixin
https://docs.djangoproject.com/en/1.8/topics/http/urls/#views-extra-options
 
To quote the relevant bits of the Django docs:

## Methods and Attributes

- **paginate_by:** An integer specifying how many objects should be displayed per page. If this is given, the view will paginate objects with paginate_by objects per page. The view will expect either a page query string parameter (via request.GET) or a page variable specified in the URLconf.

## Context

- **object_list:** The list of objects that this view is displaying. If context_object_name is specified, that variable will also be set in the context, with the same value as object_list.
- **is_paginated:** A boolean representing whether the results are paginated. Specifically, this is set to False if no page size has been specified, or if the available objects do not span multiple pages.
- **paginator:** An instance of django.core.paginator.Paginator. If the page is not paginated, this context variable will be None.
- **page_obj:** An instance of django.core.paginator.Page. If the page is not paginated, this context variable will be None

As an example of how to set up pagination in your template:

http://stackoverflow.com/questions/5907575/how-do-i-use-pagination-with-django-class-based-generic-listviews

To summarize:

"Pagination doesn't work well" - It works great!
"... isn't integrated to Django 1.7's native pagination" - It directly subclasses a Django generic view!
"... and is poorly documented." - It assumes knowledge of Django generic views and URL kwargs. I've added links to the wiki page to the relevant docs. https://github.com/etianen/django-watson/wiki/Built-in-views
